Scripts/Swift/iOSTemplateMaker/Sources/iOSTemplateMaker/SetUpCICDService.swift

```diff
@@ -33,13 +34,32 @@ struct SetUpCICDService {
 
         switch service {
         case .github:
+            var runnerType: String?
+            var runnerName: String = ""
+            while runnerType == nil {
+                print("Which workflow runner do you want to use? [(m)acos-latest/(s)elf-hosted]: ")
+                runnerType = readLine()?.lowercased()
+                if runnerType != "m" && runnerType != "s" {
+                    print("Invalid input. Please enter 'm' for macOS-latest or 's' for self-hosted.")
+                    runnerType = nil
+                }
+            }
+            // Use the selected runnerType to configure the workflow
+            if runnerType == "s" {
+                print("Configuring workflow for self-hosted runner")
+                runnerName = "[self-hosted, macOS]"
+            } else {
+                print("Configuring workflow for macOS-latest")
+                runnerName = "macOS-latest"
+            }
```

Comment:
If we create enum then we can match with enum instead like how we match `CICDService`.